tasks/config/setup-logrotate.yml

```diff
@@ -11,10 +11,25 @@
   when: ansible_facts['os_family'] == 'Debian'
 
 - name: (Amazon Linux/CentOS/Oracle Linux/RHEL) Install logrotate
+  vars:
+    distribution: "{{ ansible_facts['distribution'] }} {{ ansible_facts['system'] }} {{ ansible_facts['distribution_version'] }}"
   ansible.builtin.yum:
     name: logrotate
     state: present
-  when: ansible_facts['os_family'] == 'RedHat'
+  when:
+    - ansible_facts['os_family'] == 'RedHat'
+    - distribution != 'Amazon Linux 2023'
+
+- name: (Amazon Linux 2023) Install logrotate
+  vars:
```

Comment:
As before, we can check if whether it's Amazon Linux 2023 in the when block directly.